src/xtgeo/surface/regular_surface.py

```diff
@@ -1092,10 +1091,21 @@ def load_values(self):
 
         .. versionadded:: 2.1
         """
+
+        if self.filesrc is None:
+            raise ValueError("Can only load values into object initialised from file")
+
         if not self._isloaded:
             with warnings.catch_warnings():
                 warnings.filterwarnings("ignore", category=DeprecationWarning)
-                self.from_file(self._filesrc, fformat=self._fformat)
+
+                mfile = xtgeosys._XTGeoFile(self.filesrc)
+                kwargs = _data_reader_factory(self._fformat)(mfile, values=True)
+                self.values = kwargs.get("values", self._values)
+
+                # TODO: Sanity-checks? Should not really be necessary since instance
```

Comment:
Suggest removing comment and adding a few sanity checks in tests, for example something like this: https://github.com/equinor/xtgeo/blob/dfae6fec34f7d2f05450b0980f57d9852d7a0d32/tests/test_surface/test_file_io.py#L84 where writing the different formats and reading them back.

Reply:
> Suggest removing comment and adding a few sanity checks in tests, for example something like this:
> 
> https://github.com/equinor/xtgeo/blob/dfae6fec34f7d2f05450b0980f57d9852d7a0d32/tests/test_surface/test_file_io.py#L84
> 
> where writing the different formats and reading them back.

The sanity-checks I was considering are in `_ensure_correct_values()`, which I now see is in fact called by the `values`-setter. I'll remove the comment.

Which combos of formats are not covered in the tests? Note the check for `self.filesrc` being set: This IMO is sufficient to ensure that this lazy-loading will result in the same values as when reading the file eagerly. (Assuming the file does not change between setting up the object and lazy-loading its values, which is a case I don't think we should take respnsibility for. :smile: )

Reply:
For `_ensure_correct_values` I think it should sufice to do the same as what `from_file` and `surface_from_file` does. I think `_ensure_correct_values` might be replaced (at least partially) with some convenience in the `__init__` function.

Reply:
@oyvindeide  is this ok with you?